scripts/veloxbench/veloxbench/cpp_micro_benchmarks.py

```diff
@@ -84,10 +61,47 @@ class RecordCppMicroBenchmarks(_benchmark.Benchmark):
     iterations = 1
     flags = {"language": "C++"}
 
+    # To be overridden if using conbench
+    class conbench:
+        def mark_new_batch():
+            pass
+
+    # To be overridden if using conbench
+    def record(self, *args, **kwargs):
+        pass
+
     def run(self, **kwargs):
+        binaries = self._find_binaries(
+            REPO_ROOT.joinpath("_build", "release", "velox", "benchmarks", "basic")
+        )
         with tempfile.TemporaryDirectory() as result_dir:
-            run_command = get_run_command(result_dir, kwargs)
-            self.execute_command(run_command)
+            result_dir_path = pathlib.Path(result_dir)
+            for binary_path in binaries:
+                out_path = result_dir_path / f"{binary_path.name}.json"
+                print(
+                    f"Executing and dumping results for '{binary_path}' to '{out_path}':"
+                )
+                run_command = [
+                    binary_path,
+                    "--bm_max_secs",
+                    "10",
+                    "--bm_max_trials",
+                    "1000000",
+                    "--bm_json_verbose",
+                    out_path,
+                ]
+
+                # TODO: extend cpp micro benchmarks to allow for iterations
+                iterations = kwargs.get("iterations", None)
+                if iterations:
+                    raise NotImplementedError()
+
+                try:
+                    print(run_command)
+                    result = subprocess.run(run_command, check=True)
+                except subprocess.CalledProcessError as e:
+                    print(e.stderr.decode("utf-8"))
+                    raise e
 
             # iterate through files to make the suites
```

Comment:
should we move all this logic to open and parse the output json files to the conbench-specific library?

Reply:
Yes, that's a good idea.

Reply:
done - looks a lot cleaner, thanks!